internal/restconfig/restconfig.go

```diff
@@ -256,47 +319,95 @@ func (rcp *Producer) RunOnSelectedPrefixedContext(prefix string, function PerCon
 	return nil
 }
 
-func (rcp *Producer) AddKubeConfigFlag(cmd *cobra.Command) {
+// RunOnAllContexts runs the given function on all accessible non-prefixed contexts.
+func (rcp *Producer) RunOnAllContexts(function PerContextFn, status reporter.Interface) error {
+	if rcp.inCluster {
+		return runInCluster(function, status)
+	}
+
+	if rcp.defaultClientConfig.overrides.CurrentContext != "" {
+		// The user has explicitly chosen a context, use that only
+		return rcp.RunOnSelectedContext(function, status)
+	}
+
+	if rcp.defaultClientConfig != nil {
+		clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			rcp.defaultClientConfig.loadingRules, rcp.defaultClientConfig.overrides)
+
+		rawConfig, err := clientConfig.RawConfig()
+		if err != nil {
+			return status.Error(err, "error retrieving the raw kubeconfig setup")
+		}
+
+		if len(rcp.contexts) > 0 {
+			// Loop over explicitly-chosen contexts
+			for _, contextName := range rcp.contexts {
+				chosenContext, ok := rawConfig.Contexts[contextName]
+				if !ok {
+					status.Failure("error retrieving context %s", contextName)
+
+					continue
+				}
+
+				if err := rcp.overrideContextAndRun(chosenContext.Cluster, contextName, function, status); err != nil {
+					return err
+				}
+			}
+		} else {
+			// Loop over all accessible contexts
+			for contextName, context := range rawConfig.Contexts {
+				if err := rcp.overrideContextAndRun(context.Cluster, contextName, function, status); err != nil {
+					return err
+				}
+			}
+		}
+	}
```

Comment:
If there's no kube configs/contexts it now silently returns while, previously, `OnMultiCluster` printed a message. It also doesn't fail (see https://github.com/submariner-io/subctl/issues/196). Is this intentional?

Reply:
Right, I meant to address that later but I should do it now.